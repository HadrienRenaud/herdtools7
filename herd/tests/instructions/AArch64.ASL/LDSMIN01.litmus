AArch64 LDSMIN01
(* The smin result is zero-extended before being stored to
   teh destination register 0:X2 *)
{
int x[2] = {-1,0};
0:X1 = -128;
0:X3 = x;
int64_t 0:X2;
}

P0                 ;
LDSMIN W1, W2, [X3] ;

forall 0:X2=-1 /\ x[0]=-128 /\ x[1] = 0


