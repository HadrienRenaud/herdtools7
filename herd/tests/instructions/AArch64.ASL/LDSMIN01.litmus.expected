Test LDSMIN01 Required
States 1
0:X2=4294967295; x[0]=-128; x[1]=0;
No
Witnesses
Positive: 0 Negative: 1
Condition forall (0:X2=-1 /\ x[0]=-128 /\ x[1]=0)
Observation LDSMIN01 Never 0 1
Hash=189905288c4e5353a9326adbb28afb6b

